controllers/dns_helper.go

```diff
@@ -134,7 +125,7 @@ func (dh *dnsHelper) setEndpoints(mcgTarget *multicluster.GatewayTarget, dnsReco
 	}
 
 	sort.Slice(endpoints, func(i, j int) bool {
-		return endpoints[i].SetID() < endpoints[j].SetID()
+		return getSetID(endpoints[i]) < getSetID(endpoints[j])
```

Comment:
why did we change from SetID to getSetID ?

Reply:
We switched from out own implementation of Endpoint to an external dns one which doesn't have a `SetID()`. That said, it does have [Key](https://github.com/kubernetes-sigs/external-dns/blob/master/endpoint/endpoint.go#L305) method which we could use instead of adding  `getSetID`

Reply:
this was because it changed from our own endpoint implementation with a `setID` function, to a externaldns endpoint that doesn't have that.

Reply:
I've switched this to use Key().SetIdentifer now.

Reply:
@philbrookes You need to use the value returned by Key() as the key not just the `SetIdentifier`